### ISSUE TITLE
I've love to see MirkoPC with USB3 + 12V support

### ISSUE THREAD

**Issue by Lars-Malmgren** [mfolejewski/MirkoPC#6]
USB3 support for fast data-transfer from astro cameras.
Also support of 12V supply would give more power to M.2 module + USB3 etc.

**Comment by fanoush** [comment]
Well usb 3.0 together with native NVME means adding PCI-E switch chip to extend the single PCI-E line available - which adds a lot of complexity. OTOH I would trade more pci-e devices for that hi-fi audio in a heartbeat :-) Wonder if there is any other CM4 board with PCI-E switch and more pci-e slots builtin?

**Comment by pepe2k** [comment]
> Well usb 3.0 together with native NVME means adding PCI-E switch chip to extend the single PCI-E line available - which adds a lot of complexity. OTOH I would trade more pci-e devices for that hi-fi audio in a heartbeat :-) Wonder if there is any other CM4 board with PCI-E switch and more pci-e slots builtin?

The problem is that currently RPi bootloader doesn't support booting from NVMe behind a PCIe switch.

**Comment by fanoush** [comment]
> The problem is that currently RPi bootloader doesn't support booting from NVMe behind a PCIe switch.

Yes, they [may add it](https://www.raspberrypi.org/forums/viewtopic.php?p=1848378#p1848378) or they will not, at least they know about it and [fixed bugs related to presence of the switch](https://www.raspberrypi.org/forums/viewtopic.php?p=1838224#p1838224) so it is not completely broken and is sort of supported configuration to have the switch there. Workaround is to boot from /boot partition on something else.

I think anyone who would appreciate such board with multiple pci-e devices can workaround this issue (I have small 128MB microsd card with only the boot partition copy when booting from NVME behind switch on CM4 IO board, or it boots directly when I remove the switch from the setup)

**Comment by fanoush** [comment]
And BTW with the hypothetical board having the switch and multiple pci-e/minipcie/m2 slots you may even want booting from pci-e to sata bridge (from sata SSD, not NVME) and that config won't be supported in bootloader either so booting from microsd or emmc is necessity in such advanced configurations anyway and still such board may make sense.

Another point may be limited bandwidth of 1x PCI-E 2.0 link. However when testing with the switch and nvme SSD + VIA usb 3.0 card with sata ssd in usb3.0 to sata adapter and reading both disks concurrently the bandwidth is nicely shared over the switch and when one is not in use the other gets full speed. So I can imagine you can have e.g. another gigabit ethernet or wifi card via pci-e and still use SSD over same link without issues. One can have e.g CM4 board acting as small home NAS and gigabit router with data on sata/nvme ssd (while even possibly booting mostly read only system like opewrt from microsd/emmc). So when using internal gigabit and internal wifi and second pci-e wifi for dual channel (both 2.4 and 5GHz) and second gigabit ethernet for routing, the single 500MB/s PCI-E link may not be such bottleneck for this setup.

**Comment by pepe2k** [comment]
@fanoush 
> And BTW with the hypothetical board having the switch and multiple pci-e/minipcie/m2 slots you may even want booting from pci-e to sata bridge (from sata SSD, not NVME) and that config won't be supported in bootloader either so booting from microsd or emmc is necessity in such advanced configurations anyway and still such board may make sense.

For me, booting from microSD or eMMC is unreliable and too risky, I usually avoid such configurations. I'm aware of 2 projects with embedded PCIe switch/es (one focused on Ethernet/WiFi and one for multi 4/5G), kept on-hold because of missing support for booting from NVMe behind a switch... and of course world-wide IC/components shortage.
